src/context.ts

```diff
@@ -97,6 +101,15 @@ async function getBakeArgs(inputs: Inputs, definition: BakeDefinition, toolkit:
   await Util.asyncForEach(inputs.set, async set => {
     args.push('--set', set);
   });
+  if (await toolkit.buildx.versionSatisfies('<0.20.0')) {
+    // For buildx versions < 0.20.0, we need to set GIT_AUTH_TOKEN secret as it
+    // doesn't infer BUILDX_BAKE_GIT_AUTH_TOKEN environment variable for build
+    // request: https://github.com/docker/buildx/pull/2905
+    const gitAuthToken = getGitAuthToken(inputs);
+    if (gitAuthToken && !Bake.hasGitAuthTokenSecret(definition) && inputs.source.startsWith(Context.gitContext())) {
+      args.push('--set', `*.secrets=${Build.resolveSecretString(`GIT_AUTH_TOKEN=${gitAuthToken}`)}`);
```

Comment:
Just to be sure, this does not overwrite existing secrets, right?

How does this looks as a raw value? The actual secret value should not be argument for `--set`.

Reply:
> Just to be sure, this does not overwrite existing secrets, right?

Yes it doesn't, `!Bake.hasGitAuthTokenSecret(definition)` checks if `GIT_AUTH_TOKEN` is set first.

> How does this looks as a raw value? The actual secret value should not be argument for `--set`.

It creates a temp file setting the secret value: https://github.com/docker/bake-action/actions/runs/12707842078/job/35423580416#step:4:231

![image](https://github.com/user-attachments/assets/c4cae039-af85-48ba-9637-31a742196ff2)

And is removed when job is completed: https://github.com/docker/bake-action/actions/runs/12707842078/job/35423580416#step:6:35

![image](https://github.com/user-attachments/assets/622de3de-3e2a-48ec-b6fb-2ebcc20823f7)

This is the same behavior in build-push-action repo.

Reply:
> Yes it doesn't, !Bake.hasGitAuthTokenSecret(definition) checks if GIT_AUTH_TOKEN is set first.

I meant that if user has own secrets (not git) defined in HCL then this doesn't overwrite it. I don't remember how the merge logic was in this case.

> It creates a temp file setting the secret value: 

Temp file shouldn't be needed as buildx can just read the value from `env`, but this doesn't need to be part of this PR.

Reply:
> I meant that if user has own secrets (not git) defined in HCL then this doesn't overwrite it. I don't remember how the merge logic was in this case.

Yes we check and skip if defined either through override or within the bake definition because we solve the definition first and look at existing secrets from there: https://github.com/docker/actions-toolkit/blob/54bdcf6c08a9b43d37df4156506648c978b372fe/src/buildx/bake.ts#L411-L423

![image](https://github.com/user-attachments/assets/63ec8342-3ebd-4047-ba25-84c9a94c7674)

> Temp file shouldn't be needed as buildx can just read the value from `env`, but this doesn't need to be part of this PR.

Yes we could improve that agree, `env was not a thing before. Will open a follow-up.